packages/grafana-ui/src/types/datasource.ts

```diff
@@ -318,7 +356,9 @@ export interface DataStreamState {
   request: DataQueryRequest;
 
   /**
-   * Data may not be known yet
+   * The streaming events return entire DataFrames.  The DataSource
+   * sending the events is responcible for truncating any growing lists
+   * most likely to the requested `maxDataPoints`
```

Comment:
No idea here why this is array of dataframes and not just a dataframe. At least on Explore side this seems to be flattened anyway.

Reply:
a query often returns many DataFrames.  and again remember, the existing query request contains a *list* of queries, not a single query

Reply:
This cannot be flattened in Explore, as a time series is presented as a single DataFrame (with time and value field), and Explore needs to show more than one time series :) A single query can return many series (explore also supports multiple queries which of course can return multiple series (ie DataFrames)